pyro/poutine/trace_poutine.py

```diff
@@ -158,7 +158,10 @@ def get_trace(self, *args, **kwargs):
         Helper method for a very common use case.
         Calls this poutine and returns its trace instead of the function's return value.
         """
+        clone = kwargs.pop('clone', True)
```

Comment:
This makes me a bit uncomfortable, but I'm not sure how to work around so it seems fine for now.

Reply:
Yeah, that's true. Can we just default to returning the current trace, instead of making a defensive copy? It seems to me that it should be safe since we are anyways calling `trace_poutine` right before the return call. cc. @eb8680?

Reply:
The current behavior is better as a default because if you return a reference to `TracePoutine.trace` and then call the `TracePoutine` again, it will silently mutate the trace and may cause other problems.  If you don't want to make a defensive copy it's better to be explicit about the effectful behavior:
```python
tp = poutine.trace(fn)
tp(...)
tr = tp.trace
```

Reply:
I think it shouldn't be an issue, because this calls self, and we instantiate a new trace object in `__call__`. I will make the change to access the attribute directly, but we should eliminate the `copy` call if it is not needed. Let me know.